### PR TITLE
Convert /matrix/push to typescript

### PR DESCRIPTION
**Pull request by MidhunSureshR**

### CODE REVIEW

File: src/matrix/push/Pusher.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    pushkey: string;
+    data: IPusherData;
+    append?: boolean;
+    profile_tag?: string;
+}
+
+interface IPusherData {
+    format?: string;
+    url?: string;
+    // todo: where did this come from?
```
Comment (bwindels) [inline src/matrix/push/Pusher.ts]: This is the push data that the homeserver will pass to the notification server, sygnal, and is defined here for web push: https://github.com/matrix-org/sygnal/blob/main/docs/applications.md#push-key-and-expected-push-data

Comment (bwindels) [reply src/matrix/push/Pusher.ts]: Also see where the enablePush method in NotificationService.

**Review by bwindels — COMMENTED** [review]
lgtm, have clarified where the push data comes from. I'll leave it up to you if you want to remove the comment and/or make any other adjustments accordingly.

File: src/matrix/push/Pusher.ts (hunk 1 above)
Comment (MidhunSureshR) [reply src/matrix/push/Pusher.ts]: Thank you! Have made slight changes.

**Review by bwindels — APPROVED** [review]

**Pull request merged**